gcp_variant_transforms/libs/bigquery_schema_descriptor_test.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for infer_variant_header module."""
+
+
+from __future__ import absolute_import
+
+import unittest
+
+from apache_beam.io.gcp.internal.clients import bigquery
+from gcp_variant_transforms.libs import bigquery_schema_descriptor
+from gcp_variant_transforms.libs import bigquery_util
+
+
+class SchemaDescriptorTest(unittest.TestCase):
+  """Test case for :class:`SchemaDescriptor`."""
+
+  def setUp(self):
+    self._boolean = bigquery_util.TableFieldConstants.TYPE_BOOLEAN
+    self._float = bigquery_util.TableFieldConstants.TYPE_FLOAT
+    self._integer = bigquery_util.TableFieldConstants.TYPE_INTEGER
+    self._record = bigquery_util.TableFieldConstants.TYPE_RECORD
+    self._string = bigquery_util.TableFieldConstants.TYPE_STRING
+
+    self._nullable = bigquery_util.TableFieldConstants.MODE_NULLABLE
+    self._repeated = bigquery_util.TableFieldConstants.MODE_REPEATED
+
+  def _get_table_schema(self):
+    # type (None) -> bigquery.TableSchema
+    schema = bigquery.TableSchema()
+    schema.fields.append(bigquery.TableFieldSchema(
+        name='field_1', type=self._string, mode=self._nullable,
+        description='foo desc'))
+    schema.fields.append(bigquery.TableFieldSchema(
+        name='field_2', type=self._integer, mode=self._repeated,
+        description='foo desc'))
+    # Record field.
+    record_field = bigquery.TableFieldSchema(
+        name='record_1', type=self._record, mode=self._repeated,
+        description='foo desc')
+    record_field.fields.append(bigquery.TableFieldSchema(
+        name='record_1-field_1', type=self._boolean, mode=self._nullable,
```

Comment:
nit: I think in BigQuery, column names should not have "-" according to [this](https://cloud.google.com/bigquery/docs/schemas#column_names). I know it does not matter here since you are testing some other logic but I thought I mention since we have specific functions to convert to BQ compliant field names.

Reply:
Done!